crates/types/src/traits/block_contents.rs

```diff
@@ -177,7 +180,7 @@ pub trait BlockHeader<TYPES: NodeType>:
         builder_commitment: BuilderCommitment,
         metadata: <TYPES::BlockPayload as BlockPayload>::Metadata,
         builder_fee: BuilderFee<TYPES>,
-    ) -> impl Future<Output = Self> + Send;
+    ) -> impl Future<Output = Result<Self, Self::Error>> + Send;
```

Comment:
This is a core type to our data model, and it's not like there's a lot of places we handle the error anyway. I feel like a concrete error type makes more sense here.
In general, I agree that "anyhow for applications, concrete types for libraries" may be an overly strict rule of thumb and using anyhow in more "application"-y places of HotShot makes sense, but I'd really prefer to keep it out of the `types` crate API.

Reply:
In other words, my stance is:
With anyhow you lose in specificity for gain in convenience. I don't think it is a good tradeoff in this case, since
a) this is a trait that is going to be implemented and used by virtually any HotShot user, and there's a conceivable case wherein they may care about specific error -> reducing specificity is a relatively big trade-off
b) we don't propagate this error through long call chains or mix it with other errors in our codebase -> there isn't much convenience to be gained
c) if trait implementer (i.e. sequencer) wants to use anyhow they can just do that, because anyhow::Error satisfies all the `BlockHeader::Error` bounds.